src/decorators/restrict.js

```diff
@@ -1,9 +1,9 @@
 export default function restrict(func) {
 	return (target, key, descriptor) => {
 		const wrapped = descriptor.value
-		descriptor.value = function(...fnargs) {
-			if (func(fnargs[0])) {
-				return wrapped.apply(this, fnargs)
+		descriptor.value = async function (...fnargs){
+			if(await func(fnargs[0])){
+				wrapped.apply(this, fnargs)
 			}
 			return descriptor
```

Comment:
You should not return the descriptor here, but the wrapped function result, so the `return wrapped.apply(this, fnargs)` was a good thing :)

Reply:
okay its done :D